src/utilities/core/Logger.hpp

```diff
@@ -8,18 +8,14 @@
 
 #include "../UtilitiesAPI.hpp"
 
-#include "Singleton.hpp"
-#include "Exception.hpp"
-#include "Compare.hpp"
+#include "Compare.hpp"  // NOTE that this this is not needed for this header but so many compilation errors if I omit it
+                        // since other files transitively use Compare
```

Comment:
An unfortunate discovery. Cleaning up ours headers would be good, but that's not what I'm trying to achieve here.